iphone/Classes/TiUIiOSApplicationShortcutsProxy.m

```diff
@@ -173,19 +204,44 @@ -(void)addShortcutItem:(id)args
 
 -(UIApplicationShortcutIcon*)findIcon:(id)value
 {
+    if ([value isKindOfClass:[UIApplicationShortcutIcon class]]) {
+        return (UIApplicationShortcutIcon*)value;
+    }
+    
     if ([value isKindOfClass:[NSNumber class]]) {
         NSInteger iconIndex = [value integerValue];
         return [UIApplicationShortcutIcon iconWithType:iconIndex];
     }
     
     if ([value isKindOfClass:[NSString class]]) {
-        return [UIApplicationShortcutIcon iconWithTemplateImageName:(NSString*)value];
+        return [UIApplicationShortcutIcon iconWithTemplateImageName:[self urlInAssetCatalog:value]];
     }
     
-    NSLog(@"[ERROR] Invalid icon selection provided, defaulting to SHORTCUT_ICON_TYPE_COMPOSE");
+    NSLog(@"[ERROR] Invalid icon provided, defaulting to SHORTCUT_ICON_TYPE_COMPOSE.");
     return UIApplicationShortcutIconTypeCompose;
 }
 
+-(NSString*)urlInAssetCatalog:(NSString*)url
+{
+    NSString *resultUrl = nil;
+    
+    unsigned char digest[CC_SHA1_DIGEST_LENGTH];
+    NSData *stringBytes = [url dataUsingEncoding: NSUTF8StringEncoding];
```

Comment:
You need to strip the first slash first or `/images/myIcon.png` won't work.
See https://jira.appcelerator.org/browse/TIMOB-19709?focusedCommentId=366605&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-366605